### PR TITLE
Upgrade frameworks for jChampions Conference presentation

### PR DESCRIPTION
**Pull request by mraible**
- Spring Boot 3.2.0
- Micronaut 4.2.3
- Quarkus 3.6.4
- Helidon 4.0.2

### CODE REVIEW

**Comment by mraible** [comment]
Spring Boot 3.2.1 does not work because of [this issue](https://github.com/spring-projects/spring-boot/issues/38910), which is caused by a [bug in Spring Security](https://github.com/spring-projects/spring-security/issues/14362).

**Comment by mraible** [comment]
Helidon 4.0.2 fails with the following error when running as a native image:

```
Caused by: java.nio.charset.UnsupportedCharsetException: UTF-32LE
	at java.base@21.0.1/java.nio.charset.Charset.forName(Charset.java:543)
	at org.glassfish.json.UnicodeDetectingInputStream.<clinit>(UnicodeDetectingInputStream.java:34)
```

I entered an issue in the Helidon project at https://github.com/helidon-io/helidon/issues/8195.

File: demo.adoc [hunk 1]
```diff
@@ -401,72 +404,32 @@ Args=--initialize-at-build-time=com.okta.rest \
...
 |Framework | Command executed | Milliseconds to start
 
-|Micronaut | `./micronaut/build/native/nativeCompile/app` | 17
-|Micronaut (optimized) | `./micronaut/build/native/nativeOptimizedCompile/app` | 17.8
-|Quarkus | `./quarkus/build/quarkus-1.0.0-SNAPSHOT-runner` | 27.2
-|Spring Boot | `./spring-boot/build/native/nativeCompile/spring-boot` | 40.4
-|Helidon | `./helidon/target/helidon` | 50.4
-|Helidon (optimized) | https://github.com/oktadev/auth0-java-rest-api-examples/pull/2[auth0-java-rest-api-examples/pull/2] | 35.6
+|Micronaut | `./micronaut/build/native/nativeCompile/app` | 49.2
+|Micronaut (optimized) | `./micronaut/build/native/nativeOptimizedCompile/app` | 51.2
```
Comment (sdelamo) [inline demo.adoc:426]: Is the optimised version slower?

Comment (mraible) [reply demo.adoc:426]: Yes, that's what I'm seeing.

Comment (melix) [reply demo.adoc:426]: No, doesn't really make sense to me. However, I see that the most impactful optimizations are disabled. Not sure if disabling them is still required (optimize service loading).

**Pull request merged**